src/timescale/pull_request_github_events.service.ts

```diff
@@ -91,4 +89,156 @@ export class PullRequestGithubEventsService {
+      .andWhere(`'${startDate}'::TIMESTAMP - INTERVAL '${range} days' <= "pull_request_github_events"."event_time"`)
+      .orderBy("event_time", order);
+
+    return this.execCommonTableExpression(pageOptionsDto, cteBuilder);
+  }
+
+  async findAllWithFilters(pageOptionsDto: PullRequestPageOptionsDto): Promise<PageDto<DbPullRequestGitHubEvents>> {
+    const startDate = GetPrevDateISOString(pageOptionsDto.prev_days_start_date);
+    const range = pageOptionsDto.range!;
+    const order = pageOptionsDto.orderDirection!;
+
+    const cteBuilder = this.pullRequestGithubEventsRepository
+      .createQueryBuilder("pull_request_github_events")
+      .select("*")
+      .addSelect(`ROW_NUMBER() OVER (PARTITION BY pr_number, repo_name ORDER BY event_time ${order}) AS row_num`)
+      .orderBy("event_time", order);
+
+    /*
+     * todo: filter for spam PRs.
+     * The way this was done in v1 was to check for any and all PRs in repos
+     * that had a "spam" like label on them. As of this writing, the events feed
+     * nozzle does not capture labels for PRs / issues.
+     */
+
+    /* filter for hacktoberfest PRs */
+    switch (pageOptionsDto.topic) {
+      case "hacktoberfest":
+        cteBuilder
+          .where(this.hacktoberfestPrFilterBuilderStart())
+          .andWhere(this.hacktoberfestPrFilterBuilderEnd(range));
+        break;
+      default:
+        cteBuilder
+          .where(`'${startDate}'::TIMESTAMP >= "pull_request_github_events"."event_time"`)
+          .andWhere(
+            `'${startDate}'::TIMESTAMP - INTERVAL '${range} days' <= "pull_request_github_events"."event_time"`
+          );
+        break;
+    }
+
+    /* filter on PR author / contributor */
+    if (pageOptionsDto.contributor) {
+      cteBuilder.andWhere(`"pull_request_github_events"."pr_author_login"=:author`, {
+        author: pageOptionsDto.contributor,
+      });
+    }
+
+    /*
+     * apply repo specific filters (topics, top 100, etc.) - this captures a few
+     * pre-defined filters provided by the PullRequestPageOptionsDto.
+     * This will call out to the API connection to get metadata on the repos.
+     */
+    if (pageOptionsDto.filter || pageOptionsDto.topic) {
+      const filtersDto: RepoSearchOptionsDto = {
+        filter: pageOptionsDto.filter,
+        topic: pageOptionsDto.topic,
+        limit: 50,
+        skip: 0,
+        range,
+      };
+
+      const repos = await this.repoService.findAllWithFilters(filtersDto);
+      const repoNames = repos.data.map((repo) => repo.full_name);
+
+      cteBuilder.andWhere(`"pull_request_github_events"."repo_name" IN (:...repoNames)`, {
+        repoNames,
+      });
+    }
+
+    /* apply user provided repo name filters */
+    if (pageOptionsDto.repo) {
+      cteBuilder.andWhere(`"pull_request_github_events"."repo_name" IN (:...repoNames)`, {
+        repoNames: pageOptionsDto.repo.split(","),
+      });
```

Comment:
This should be just one unless we are building in support to expand it into multiples for later.

```suggestion
      cteBuilder.andWhere(`"pull_request_github_events"."repo_name" =:repo`, {
        repoNames: pageOptionsDto.repo,
      });
```

Reply:
This definitely works for multiple repo names. For example:

```
curl -X 'GET' \
  'http://localhost:3003/v2/prs/search?page=1&limit=10&range=30&prev_days_start_date=0&repo=open-sauced%2Fapp%2Copen-sauced%2Fapi' \
  -H 'accept: application/json'
```

which sets repo is set as:
```
open-sauced/app,open-sauced/api
```

returns results for both repos.